### DISCUSSION TITLE
[Help wanted] how to get user identity from customized event

### DISCUSSION THREAD

**Discussion by archerbj** [plausible/analytics#1141] [Self-Hosted Support]
Hi, Thanks for the great project first!

I am trying to figure out how to record user actions on my website. And the customized event + goal can do the work.

But from the docs, customized event should not carry any personal identity information like username.

>Note that you must ensure that no personally identifiable information (PII) is sent to Plausible Analytics with custom properties. PII is information that could be used on its own to identify, contact, or precisely locate an individual.

If this is the rule, how can I know which user made the action?
Any may I ask why PII shouldn't be sent to Plausible Analytics ?

**Comment by metmarkosaric** [comment]
Plausible is built to be privacy-first and compliant with the different regulations [such as GDPR](https://plausible.io/blog/google-analytics-gdpr). We believe you can get all the insights you need to run your site/business without needing to track individual people and by simply looking at the aggregate numbers.

**Reply by TheLastGimbus** [reply]
Hi!

Tried out plausible on my website, and very like it so far! I was thinking about adding it to my mobile flutter apps, but i'm seeing the problem with this: if there will be new user_id for every new ip (and on mobile, that is - *frequent*) - i will see much more unique users than there actually are :(

i could generate unique pure-random uuid at the app start and use that... however this *mayyy* be a bit privacy-violating if i'm gonna collect *lot* of data :thinking:  (a lot => one year of activity => potentially fingerprinting the guy :thinking: )

maybe generating it every 24 hours would be good? :raised_eyebrow:

**Reply by subpublic** [reply]
I'm a little confused about PII when tracking a logged in user. If we'd store a randimized GUID as user_id (meaningless from outside) and that only the data owner (our customer) would be allowed to map that user_id to real identification and tracking.
Would that be considered a GDPR violation?

**Reply by andrashann** [reply]
@metmarkosaric What if people can post their own content on the website – can I still not send the _author_ (not the viewer!) username with the page view? This information would be available anyway as the username is part of the post url, e.g. `example.com/john/my-favorite-jazz-songs` (john is a user who selected this name during registration, it may or may not be a real name). 

In your own example you mention _"For example, say you want to track your blog posts by author."_, but it is not clear if it is only okay if this is published like a newspaper (i.e. a given set of "official" authors) or also with user generated content.
